### ISSUE TITLE
lxqt-panel: clock plugin: scale font

### ISSUE THREAD

**Issue by Vladimir-csp** [lxde/lxqt#660]
When panel is vertical, but clock's rotation is disabled, try to scale it to fit the width of panel.

**Comment by jleclanche** [comment]
No, that's not just really hard to do, it's pretty much impossible in most cases where the panel is already thin.

**Comment by Vladimir-csp** [comment]
without that the clock just gets clipped. In some cases tiny font is better than clipped text.
If it is hard to do technically, then never mind.

**Comment by jleclanche** [comment]
The clock shouldn't get clipped, the panel should be wide enough to support the largest the clock can get.

**Comment by jleclanche** [comment]
cc @paulolieuthier @palinek

**Comment by jleclanche** [comment]
I don't think this should be solved at a technical level, we should just ensure the clock has some padding to it.

**Comment by Vladimir-csp** [comment]
But if user does not want panel thicker than X pixels? Maybe it is impractical to squeeze year in here, but in case of a bit thicker panel, it could be with the cost of a bit smaller font:
![panel](https://cloud.githubusercontent.com/assets/4061903/7906910/da21fe36-083a-11e5-918b-fe973272400f.png)

By the way the clock here unclipped and centered only after panel restart. If you set shorter format, clock center is not updated on the fly.

**Comment by pmattern** [comment]
This is closely related if not even a duplicate of #632.

**Comment by Vladimir-csp** [comment]
I do not see where #632 is calling for autoadjusting panel width. And IMHO, this is a bad idea, makes panel unpredictable.

**Comment by jleclanche** [comment]
It's the same root issue, it calls for adjusting the *plugin* width.

**Comment by Vladimir-csp** [comment]
OK, then merge: there are two logical actions:
- adjust plugin size if it is possible (not bumping into panel limits) 
- scale down plugin content if it's size can not be increased.

Although, it feels like lots of redundant margins and paddings are at play on #632 screenshots.

**Comment by pmattern** [comment]
@Vladimir-csp 
Just edited my last post to avoid further misunderstanding.

**Comment by Vladimir-csp** [comment]
I did not call for adjusting panel width automatically, maybe I've used some vague phrasing?
On contrary, this issue is about scaling down plugin content, i.e. reducing clock font. Just like taskbar buttons scale down when there are lots of them spammed together.

**Comment by kuzmas** [comment]
@Vladimir-csp: I think you could use the worldclock plugin here. It's more flexible and allows you to set (using html) whatever format and font size you like.

**Comment by Vladimir-csp** [comment]
Thanks for the suggestion! That's a nice plugin.
I thought that if auto scaling is already done for taskbar plugin, then the same mechanism or something similar could be applied to other plugins to avoid clipping due mismatch with panel size (not padding issues).

**Comment by Vladimir-csp** [comment]
By the way, world clock also a bit slow at applying format changes. I takes up to half a minute and sometimes center the content in a weird way.
